1_fetch/src/check_characteristics.R

```diff
@@ -0,0 +1,63 @@
+#' Function to read all valid characteristic names from WQP web service
+#' 
+#' @return vector of character strings representing all of the valid
+#' characteristic names from WQP
+#' 
+read_wqp_characteristics <- function(){
+  
+  # Fetch all characteristics from WQP
+  all_characteristic_names <- xml2::read_xml("https://www.waterqualitydata.us/Codes/Characteristicname?mimeType=xml") %>% 
+    xml2::xml_children() %>% xml2::xml_attr("value") %>% na.omit()
+  
+  return(all_characteristic_names)
+}
+
+
+#' Function to check whether all user-specified characteristics are valid.
+#' 
+#' @param chars vector of character strings representing which characteristic
+#' names to check against valid entries in the WQP.
+#' 
+#' @return logical statement for each string within supplied characteristics list;
+#' if any characteristics are not valid, a printed message will indicate which 
+#' characteristics do not exist in WQP. 
+#' @example check_valid_characteristics(c("Specific conductivity", "Specific conductance"))
+#' 
+check_valid_characteristics <- function(chars) {
+  
+  # Fetch all WQP characteristics and test whether each user-supplied characteristic
+  # name is within the list of valid entries from WQP
+  all_characteristic_names <- read_wqp_characteristics()
+  chars_exist <- chars %in% all_characteristic_names
+  
+  if(!all(chars_exist)) {
+    message(sprintf("The following characteristics do not exist in WQP:\n\n%s\n",
```

Comment:
We might want to consider using `warning()` instead of just `message()` here - it carries a bit more weight when users are reading console output.

Reply:
It shows up at the end of all the output instead of in the middle, so we would end up with 

```r
> check_valid_characteristics(c("Specific conductivity", "Specific conductance bad"))
[1]  TRUE FALSE
Warning message:
In check_valid_characteristics(c("Specific conductivity", "Specific conductance bad")) :
  The following characteristics do not exist in WQP:

Specific conductance bad
```

instead of 
```
> check_valid_characteristics(c("Specific conductivity", "Specific conductance bad"))
The following characteristics do not exist in WQP:

Specific conductance bad

[1]  TRUE FALSE
```